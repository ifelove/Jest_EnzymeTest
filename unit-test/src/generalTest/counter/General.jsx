import React from "react";

const General = () => {
  const [counter, setCounter] = React.useState(1);

  const decrease = (prev) => {
setCounter( prev - 1);
  };

  const increase = (prev) => {
 setCounter (prev + 1);
  };

  return (
    <div>
      <h5>Counter App</h5>

      <div>
        {" "}
        <button data-testid="decrement" onClick={() => decrease(counter)}>
          Prev{" "}
        </button>
        <span data-testid="counter"> {counter}</span>
        <button data-testid="increment" onClick={() => increase(counter)}>
          Next{" "}
        </button>
      </div>
    </div>
  );
};

export default General;
