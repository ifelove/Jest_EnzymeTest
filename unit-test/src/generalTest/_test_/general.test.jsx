import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer"
import General from "../counter/General";

describe("testing the the counter", () => {
  test("increament testing", () => {
    //rendering the component in virtual dom
    render(<General />);

    // select the element to test
    const counter = screen.getByTestId("counter");
    // const decrement = screen.getByTestId("decrement");
    const incrementBtn = screen.getByTestId("increment");

    fireEvent.click(incrementBtn);

    expect(counter).toHaveTextContent("2");
  });

  test("decreament testing", () => {
    //rendering the component in virtual dom
    render(<General />);

    // select the element to test
    const counter = screen.getByTestId("counter");
    // const decrement = screen.getByTestId("decrement");
    const decrementBtn = screen.getByTestId("decrement");

    fireEvent.click(decrementBtn);

    expect(counter).toHaveTextContent("0");
  });
});

describe(" counter snapshot", () => {


    it("It show matches DOM snapshot",()=>{

        const tree=renderer.create(<General/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
});
