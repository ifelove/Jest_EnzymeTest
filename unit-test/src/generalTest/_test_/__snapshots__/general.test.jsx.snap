// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` counter snapshot It show matches DOM snapshot 1`] = `
<div>
  <h5>
    Counter App
  </h5>
  <div>
     
    <button
      data-testid="decrement"
      onClick={[Function]}
    >
      Prev
       
    </button>
    <span
      data-testid="counter"
    >
      1
    </span>
    <button
      data-testid="increment"
      onClick={[Function]}
    >
      Next
       
    </button>
  </div>
</div>
`;
